catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/NewDataDialogTest.java

```diff
@@ -0,0 +1,220 @@
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testInvalidVarNames() {
+		String emptyVarName = "";
+		String spaceVarName = " ";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(emptyVarName);
+
+		clickOk();
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+
+		onView(withId(R.id.input_edit_text))
+				.perform(typeText(spaceVarName));
+
+		clickOk();
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testInvalidThenValidVarNameErrorMessage() {
+		String spaceVarName = " ";
+		String validVarName = "name";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(spaceVarName);
+
+		checkErrorMessageDisplayed(R.string.name_consists_of_spaces_only);
+
+		onView(withId(R.id.input_edit_text))
+				.perform(typeText(validVarName));
+
+		clickOk();
+
+		checkVariableShownInSpinner(validVarName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testValidSpecialCharVarNames() {
+		String underscoreVarName = "var_var";
+		String numberVarName = "var123";
+		String validSpecialCharVarName = "var$";
+		String slashVarName = "var/";
+		String bracketVarName = "var(";
+		String curlyBracketVarName = "var{";
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariableInitial(underscoreVarName);
+		checkVariableShownInSpinner(underscoreVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(numberVarName);
+		checkVariableShownInSpinner(numberVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(validSpecialCharVarName);
+		checkVariableShownInSpinner(validSpecialCharVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(slashVarName);
+		checkVariableShownInSpinner(slashVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(bracketVarName);
+		checkVariableShownInSpinner(bracketVarName);
+
+		onBrickAtPosition(variableBrickPosition).onVariableSpinner(R.id.set_variable_spinner)
+				.performNewVariable(curlyBracketVarName);
+		checkVariableShownInSpinner(curlyBracketVarName);
+	}
+
+	@Category({Cat.AppUi.class, Level.Functional.class})
+	@Test
+	public void testCancelVarCreation() {
+		clickOnEmptySpinner(variableBrickPosition, R.id.set_variable_spinner);
+
+		clickCancel();
+
+		onBrickAtPosition(variableBrickPosition)
+				.checkShowsText(R.string.brick_variable_spinner_create_new_variable);
+	}
+
+	private void clickOnEmptySpinner(int brickPosition, int spinnerId) {
```

Comment:
since method name is misleading (it clicks on the spinner, it doesnt check that its empty, its just something you have to know and take care of outside the method), the parameters arent necessarily clear, and theres only 3 occurences.
please remove this method and use the chunk of code in place. (its only three occurences anyway).
id rather be trading little code duplication for way better readability.